README.md

```diff
@@ -1,25 +1,115 @@
 # glazier
 
-`glazier` is an attempt to provide a common interface to the various
-elements of different platform application frameworks. It is designed to be used
-by [druid], a UI toolkit.
+Glazier is a portable infrastructure layer intended for high quality GUI
+toolkits in Rust. It is based on druid-shell but with some changes, primarily
+around making it agnostic to the choice of drawing.
+
+Primary platforms are Windows, macOS, and Linux. Web is a secondary platform.
+Other ports may happen if they are contributed.
+
+This library is currently work in progress and should be considered experimental.
+Contributions are welcome, see [CONTRIBUTING](./CONTRIBUTING.md) for more details.
+
+## Scope
+
+The following tasks are in scope. Mostly they are implemented, but as always
+there is more refinement to be done.
+
+* Window creation, including subwindows.
+
+* System menus. These are especially important on macOS.
+
+* Keyboard events.
+
+* Input Method Editor. On macOS, correct handling of dead keys is through the
+IME, and an application built on Glazier is expected to handle IME.
+
+* Mouse and pointer events. Mouse events are currently supported, pointers
+are a hoped-for feature.
+
+* Providing DPI information to the application.
```

Comment:
Ooh, this is interesting. My impression from quick research is that this is impossible in the general case (consider e.g. a projector).

I would expect the much more standard approach of scaling factor to be provided instead- it seems quite error prone for applications to try to be clever with DPI information.

(Obviously there are a couple of cases (I guess only real-size for printing or other 2d items; and I guess an orthographic project of 3d items?) where this could make sense)

Reply:
For Wayland it's unclear where the protocol extensions will sit for fractional scaling.

Reply:
By DPI scaling I mean the logical pixel / device pixel ratio. I use the term "logical pixel" here to refer to the concept called "px" in CSS, "point" in macOS, "dp" in Android, and probably lots of other terms in other systems. I don't mean physical DPI, as that's not really practical, but the word "dpi" and "high dpi" are widely used for this.